gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/RpcBatch.java

```diff
@@ -28,37 +32,79 @@
 public interface RpcBatch {
 
   /**
+   * An interface for batch callbacks.
+   */
+  interface Callback<T> {
+
+    /**
+     * This method will be called upon success of the batch operation.
+     */
+    void onSuccess(T response);
+
+    /**
+     * This method will be called upon failure of the batch operation.
+     */
+    void onFailure(GoogleJsonError googleJsonError);
+  }
+
+  /**
    * Adds a call to "list zones" to the batch with the provided {@code callback} and {@code
    * options}.
    */
-  void addListZones(DnsRpc.Callback<ManagedZonesListResponse> callback,
-      Map<DnsRpc.Option, ?> options);
+  void addListZones(Callback<ManagedZonesListResponse> callback, Map<DnsRpc.Option, ?> options);
 
   /**
    * Adds a call to "create zone" to the batch with the provided {@code callback} and {@code
    * options}.
    */
-  void addCreateZone(ManagedZone zone, DnsRpc.Callback<ManagedZone> callback,
+  void addCreateZone(ManagedZone zone, Callback<ManagedZone> callback,
       Map<DnsRpc.Option, ?> options);
 
   /**
    * Adds a call to "get zone" to the batch with the provided {@code callback} and {@code options}.
    */
-  void addGetZone(String zoneName, DnsRpc.Callback<ManagedZone> callback,
-      Map<DnsRpc.Option, ?> options);
+  void addGetZone(String zoneName, Callback<ManagedZone> callback, Map<DnsRpc.Option, ?> options);
 
   /**
    * Adds a call to "delete zone" to the batch with the provided {@code callback} and {@code
    * options}.
    */
-  void addGetProject(DnsRpc.Callback<Project> callback,
-      Map<DnsRpc.Option, ?> options);
+  void addGetProject(Callback<Project> callback, Map<DnsRpc.Option, ?> options);
 
   /**
    * Adds a call to "get project" to the batch with the provided {@code callback} and {@code
    * options}.
    */
-  void addDeleteZone(String zoneName, DnsRpc.Callback<Void> callback);
+  void addDeleteZone(String zoneName, Callback<Void> callback);
+
+  /**
+   * Adds a call to "list record sets" to the batch with the provided {@code callback} and {@code
```

Comment:
Do you think it's worth mentioning the `zoneName` parameter? Also in the following 3 methods.